### PR TITLE
Guide for theming with `styled-component` with `typescript`

### PR DESCRIPTION
**Pull request by marsinearth**
now I fixed the git rebase issue!

### CODE REVIEW

**Comment by marsinearth** [comment]
styled-components theming complete!

**Comment by hyochan** [comment]
@marsinearth  The snapshot test is failing please update snapshot by `jest -u`

**Comment by marsinearth** [comment]
> @mars-lan The snapshot test is failing please update snapshot by `jest -u`

that's a wrong person you mentioned btw ;) I'm trying to solve the problem using react-native-testing-library and figuring out how to mock the animated.loop. Can you guide me about abc of jest.mock? have no idea. for react-native-animatable text opacity issue, which is the main reason of failure of the circleci test, is solved by putting async/await on loading.test.ts with waitForElement of react-native-testing-library but now I'm having issue of snapshot images are not matching with the rotate angle..

**Comment by hyochan** [comment]
Ah.. sorry for confusion on naming, my mistake 😅. I've changed it.
I will try to look over your codes.

**Comment by hyochan** [comment]
@marsinearth Yo it's working fine in my env. Haven't changed anything.
image


@marsinearth Could you add below code in `jestSetup.ts` and try again?
```
import { NativeModules, Animated } from 'react-native';
...
// at the very bottom
Object.defineProperty(Animated, 'start', {value: () => jest.fn()});
```

**Comment by hyochan** [comment]
@marsinearth Yeah you've made it! 👍I'll pend this from merging until `Wed`.

File: __mocks__/react-native-animatable.tsx [hunk 1]
```diff
@@ -1,9 +1,10 @@
+
 import React from 'react';
 import { Image, Text } from 'react-native';
 
 const createAnimatableComponent = (Component) => (
   class AnimatableComponent extends React.Component {
-    public render() {
+    render() {
```
Comment (hyochan) [inline __mocks__/react-native-animatable.tsx]: Please revert this changes
